chainer/functions/batch_normalization.py

```diff
@@ -130,6 +132,35 @@ def __call__(self, x, test=False, finetune=False):
         self.is_finetune = finetune
         return function.Function.__call__(self, x)
 
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 1)
+        x_type, = in_types
+
+        if isinstance(self.size, tuple):
+            feature_shape = type_check.Variable(self.size, 'feature_shape')
+            feature_ndim = len(self.size)
+        else:
+            feature_shape = type_check.Variable((self.size,), 'feature_shape')
+            feature_ndim = 1
+
+        type_check.expect(
+            x_type.dtype == numpy.float32,
+            x_type.ndim >= type_check.Variable(
+                feature_ndim, 'feature_ndim') + 1,
```

Comment:
Combined with the above comment, how about using `self_.size.__len__()` instead of the `feature_ndim` variable?